File: Dot examples/ASP Book Examples/Book Examples/Chapter34/cs/App_Code/Entities/Book.cs
using System;
using System.Data.Linq.Mapping;

/// <summary>
/// Summary description for Book
/// </summary>
public partial class Book: EntityBase<Book>
{

    protected override void Validate()
    {

        if (!ValidationUtility.SatisfiesRequired(Title))
            ValidationErrors.Add("Title", "Title is required", "required");
        if (!ValidationUtility.SatisfiesRequired(Authors))
            ValidationErrors.Add("Authors", "Authors is required", "required");
        if (!ValidationUtility.SatisfiesRequired(Description))
            ValidationErrors.Add("Description", "Description is required", "required");
        if (!ValidationUtility.SatisfiesType(BuyLink, "url"))
            ValidationErrors.Add("BuyLink", "Buy link is invalid URL", "invalid");

    }



}

File: Dot examples/ASP Book Examples/Book Examples/Chapter34/vb/App_Code/Entities/Book.vb
Imports System
Imports System.Data.Linq.Mapping

''' <summary>
''' Summary description for Book
''' </summary>
Partial Public Class Book
     Inherits EntityBase(Of Book)

    Protected Overrides Sub Validate()

        If Not ValidationUtility.SatisfiesRequired(Title) Then
            ValidationErrors.Add("Title", "Title is required", "required")
        End If
        If Not ValidationUtility.SatisfiesRequired(Authors) Then
            ValidationErrors.Add("Authors", "Authors is required", "required")
        End If
        If Not ValidationUtility.SatisfiesRequired(Description) Then
            ValidationErrors.Add("Description", "Description is required", "required")
        End If
        If Not ValidationUtility.SatisfiesType(BuyLink, "url") Then
            ValidationErrors.Add("BuyLink", "Buy link is invalid URL", "invalid")
        End If

    End Sub



End Class

